src/vivarium/framework/results/observer.py

```diff
@@ -0,0 +1,64 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Dict
+
+import pandas as pd
+
+from vivarium import Component
+from vivarium.framework.engine import Builder
+
+
+class Observer(Component, ABC):
+    """An abstract base class intended to be subclassed by observer components.
+    The primary purpose of this class is to provide attributes required by
+    the subclass `report` method.
+
+    Note that a `register_observation` method must be defined in the subclass.
+    """
+
+    @abstractmethod
+    def register_observations(self, builder: Builder) -> None:
+        """(Required). Register observations with within each observer."""
+        pass
+
+    def setup_component(self, builder: Builder) -> None:
+        super().setup_component(builder)
+        self.register_observations(builder)
+        self.get_report_attributes(builder)
+
+    def get_report_attributes(self, builder: Builder) -> None:
+        """Define commonly-used attributes for reporting."""
+        try:
+            self.results_dir = builder.configuration.output_data.results_directory
+        except KeyError:
+            self.results_dir = None
+        self.input_draw = builder.configuration.input_data.input_draw_number
+        self.random_seed = builder.configuration.randomness.random_seed
```

Comment:
What happens if you have a simulation with a different ArtifactManager that doesn't have draws?

I'd suggest this:
```suggestion
    def __init__(self):
        super().__init__()
        self.results_dir = None
        self.input_draw = None
        self.random_seed = None
    
    def get_report_attributes(self, builder: Builder) -> None:
        """Define commonly-used attributes for reporting."""
        if "results_directory" in builder.configuration.output_data.:
            self.results_dir = builder.configuration.output_data.results_directory
        if "input_draw_number" in builder.configuration.input_data:
            self.input_draw = builder.configuration.input_data.input_draw_number
        if "random_seed" in builder.configuration.randomness:
            self.random_seed = builder.configuration.randomness.random_seed
```

Reply:
This doesn't play well due to the values being nested two levels deep. How about the changes I've pushed using `.to_dict()` and `.get()`?